trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/rest/ParsedS3Request.java

```diff
@@ -13,13 +13,17 @@
  */
 package io.trino.s3.proxy.server.rest;
 
+import io.trino.s3.proxy.server.signing.ParsedAuthorization;
 import jakarta.ws.rs.core.MultivaluedMap;
 
 import java.util.Optional;
 
 import static java.util.Objects.requireNonNull;
 
 public record ParsedS3Request(
+        ParsedAuthorization parsedAuthorization,
+        Optional<String> session,
+        String xAmzDate,
```

Comment:
Can we find a better name? I know it matches the header but it looks kind of ugly here.

Reply:
What about `requestDate`? Particularly since we may want to load this from `x-amz-date` _OR_ the `Date` header ([docs](https://docs.aws.amazon.com/AmazonECR/latest/APIReference/CommonParameters.html))